examples/visualization/plot_publication_figure.py

```diff
@@ -165,3 +179,90 @@
 for ax, label in zip(axes, 'AB'):
     ax.text(0.03, ax.get_position().ymax, label, transform=fig.transFigure,
             fontsize=12, fontweight='bold', va='top', ha='left')
+
+###############################################################################
+# Custom timecourse with montage inset
+# ------------------------------------
+#
+# Suppose we want a figure with some mean timecourse extracted from a number of
+# sensors, and we want a smaller panel within the figure to show a head outline
+# with the positions of those sensors clearly marked.
+# If you are familiar with MNE, you know that this is something that
+# :func:`mne.viz.plot_compare_evokeds` does, see an example output in
+# :ref:`ex-hf-sef-data` at the bottom.
+#
+# In this part of the example, we will show you how to achieve this result on
+# your own figure, without having to use :func:`mne.viz.plot_compare_evokeds`!
+#
+# Let's start by loading some :ref:`example data <sample-dataset>`.
+
+data_path = mne.datasets.sample.data_path()
+fname_raw = op.join(data_path, "MEG", "sample", "sample_audvis_raw.fif")
+raw = mne.io.read_raw_fif(fname_raw)
+
+# For the sake of the example, we focus on EEG data
+raw.pick_types(meg=False, eeg=True)
+
+
+###############################################################################
+# Let's make a plot.
+
+# channels to plot:
+to_plot = [f"EEG {i:03}" for i in range(1, 5)]
+
+# get the data for plotting in a short time interval from 10 to 20 seconds
+start = int(raw.info['sfreq'] * 10)
+stop = int(raw.info['sfreq'] * 20)
+data, times = raw.get_data(picks=to_plot,
+                           start=start, stop=stop, return_times=True)
+
+# Scale the data from the MNE internal unit V to µV
+data *= 1e6
+
+# Take the mean of the channels
+mean = np.mean(data, axis=0)
+
+# make a figure
+fig, ax = plt.subplots(figsize=(4.5, 3))
+
+# plot some fake EEG data
+ax.plot(times, mean)
+ax.set(xlabel="Time (s)", ylabel="Amplitude (µV)")
+
+###############################################################################
+# So far so good. Now let's add the smaller figure within the figure to show
+# exactly, which sensors we used to make the timecourse.
+
+# For that, we use an "inset_axes" that we plot into our existing axes
+axins = inset_locator.inset_axes(ax, width="30%", height="30%", loc=2)
+
+# The head outline with the sensor positions can be plotted using the
+# MNE :class:`mne.io.raw` object that is the origin of our data.
+# Specifically, that object already contains all the sensor positions,
+# and we can plot them using the ``plot_sensors`` method.
+raw.copy().pick_channels(to_plot).plot_sensors(
```

Comment:
do we need the `.copy()` here?

Reply:
because `.pick_channels` works in-place and would change our `raw`, which a user may not want

Reply:
I agree that in an interactive session that's probably what one would do, and it's good for model code to reflect actual practice as much as possible. But for the doc-building server, memory usage can be an issue (it's more strict when building/deploying the full docs after merge, than it is for building in-progress PRs).  *Probably* this won't be an issue since you're already restricting to just EEG channels, but in general be aware when writing examples/tutorials to be conservative about memory.

In this case I'd be inclined to take one of two paths:

1. remove the copy and add a comment saying "pick_channels edits the raw object in place, so normally we'd make a copy first in case we needed to re-use the raw with all sensors intact in later code. Here, all we need it for is plotting these 5 sensors, so we won't bother"

2. leave the copy in, but explain why ("pick_channels edits the raw object in place, so we'll make a copy here so that our raw object stays intact for later analysis"). Although that's not *strictly* true for this tutorial, it's likely to be true in general.

Personally I lean toward # 2

Reply:
Thanks, I agree that it's good to better explain *why* a copy is being made.